packages/nimble-components/src/table-column/icon/tests/table-column-icon.stories.ts

```diff
@@ -0,0 +1,152 @@
+import { tableColumnIconTag } from '..';
+import {
+    SharedTableArgs,
+    sharedTableArgTypes,
+    sharedTableArgs
+} from '../../base/tests/table-column-stories-utils';
+import { iconXmarkTag } from '../../../icons/xmark';
+import { tableColumnTextTag } from '../../text';
+import { iconCheckLargeTag } from '../../../icons/check-large';
+import { iconQuestionTag } from '../../../icons/question';
+import { mappingIconTag } from '../../../mapping/icon';
+
+const simpleData = [
+    {
+        firstName: 'Ralph',
+        lastName: 'Wiggum',
+        status: 'fail',
+        isChild: true
+    },
+    {
+        firstName: 'Marge',
+        lastName: 'Simpson',
+        status: 'success',
+        isChild: false
+    },
+    {
+        firstName: 'Homer',
+        lastName: 'Simpson',
+        status: 'success',
+        isChild: false
+    },
+    {
+        firstName: 'Bart',
+        lastName: 'Simpson',
+        status: 'success',
+        isChild: true
+    },
+    {
+        firstName: 'Greg',
+        lastName: '',
+        status: 'foo',
+        isChild: true
+    },
+    {
+        firstName: 'Frank',
+        lastName: '',
+        status: 'bar',
+        isChild: false
+    }
+];
+
+const metadata: Meta<IconColumnTableArgs> = {
+    title: 'Table Column - Icon'
+};
+
+export default metadata;
+
+interface IconColumnTableArgs extends SharedTableArgs {
+    fieldName: string;
+    keyType: string;
+    checkValidity: () => void;
+    validity: () => void;
+}
+
+const iconColumnDescription = 'The `nimble-table-column-icon` column renders string, number, or boolean values as a Nimble icon or `nimble-spinner` in the `nimble-table`.';
+
+const validityDescription = `Readonly object of boolean values that represents the validity states that the column's configuration can be in.
+The object's type is \`TableColumnValidity\`, and it contains the following boolean properties:
+
+-   \`invalidMappingKeyValueForType\`: \`true\` a mapping has a \`key\` that is not of the \`key-type\` declared by the column
+-   \`multipleDefaultMappings\`: \`true\` when multiple mappings have the \`default-mapping\` attribute
+-   \`unsupportedMappingType\`: \`true\` when the column contains a mapping element other than \`nimble-mapping-text\`
+-   \`duplicateMappingKey\`: \`true\` when multiple mappings have the same \`key\` value
+-   \`missingKeyValue\`: \`true\` when a mapping has no \`key\` value, and it is not marked with \`default-mapping\`
+-   \`invalidIconName\`: \`true\` when a mapping's \`icon\` value is not the tag name of a valid, loaded Nimble icon (e.g. \`nimble-icon-check\`)
+`;
+
+export const iconColumn: StoryObj<IconColumnTableArgs> = {
+    parameters: {
+        docs: {
+            description: {
+                story: iconColumnDescription
+            }
+        }
+    },
+    // prettier-ignore
+    render: createUserSelectedThemeStory(html<IconColumnTableArgs>`
+        ${usageWarning('table')}
+        <${tableTag}
```

Comment:
Any reason not to add a spinner mapping to this example?

Reply:
I just didn't know how to incorporate it in a meaningful way into the example, but it doesn't really have to make sense, so I'll add it.